src/v/cluster/rm_stm.h

```diff
@@ -414,12 +419,20 @@ class rm_stm final : public persisted_stm {
         // conflicts. if the replication fails we reject a command but clients
         // by spec should be ready for thier commands being rejected so it's
         // ok by design to have false rejects
-        absl::flat_hash_map<model::producer_identity, seq_entry> seq_table;
+        absl::node_hash_map<model::producer_identity, seq_entry_wrapper>
+          seq_table;
 
         absl::flat_hash_map<model::producer_identity, model::tx_seq> tx_seqs;
         absl::flat_hash_map<model::producer_identity, expiration_info>
           expiration;
 
+        // We need to store replication order for idempotent pids only. So we
+        // will use intrusive list for it.
+        using idempotent_pids_replicate_order = counted_intrusive_list<
+          seq_entry_wrapper,
+          &seq_entry_wrapper::_hook>;
+        idempotent_pids_replicate_order idempotent_pids_order;
```

Comment:
can we name it to something like `lru_pids` to highlight where the order comes from and add a comment to clarify the order e.g. first element is the least recently used and and last element is the recently updated.